### ISSUE TITLE
aiohttp error when running flows

### ISSUE THREAD

**Issue by bjorhn** [PrefectHQ/prefect-azure#101]
### Error description
The changes in #97 have resulted in the following error on the agent when trying to run a flow:
`ImportError: aiohttp package is not installed`

### Proposed solution
I think this should be fixed by adding the aiohttp package to prefect-azure's requirements.txt. It's already in requirements-dev.txt, which explains why this wasn't caught during development. I'm currently unable to test this properly, otherwise I would have created a PR.

This is what azure-identity has to say about using azure.identity.aio:

> This library includes a set of async APIs. To use the async credentials in [azure.identity.aio](https://aka.ms/azsdk/python/identity/aio/docs), you must first install an async transport, such as [aiohttp](https://pypi.org/project/aiohttp/).

### Workaround
We are currently attempting to work around the problem by explicitly adding aiohttp to our agent's list of requirements.

This seems to solve the above issue, however we run into another issue instead which I think is unrelated but which we also haven't seen before upgrading from 0.2.6 to 0.2.8:
```09:46:50.965 | INFO    | prefect.agent - Submitting flow run '244d6518-245a-49ae-a465-7ec4a9437616'
09:46:53.772 | ERROR   | prefect.agent - Failed to submit flow run '244d6518-245a-49ae-a465-7ec4a9437616' to infrastructure.
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/prefect/agent.py", line 500, in _submit_run_and_capture_errors
    result = await infrastructure.run(task_status=task_status)
  File "/usr/local/lib/python3.8/site-packages/prefect_azure/container_instance.py", line 336, in run
    container_group = self._configure_container_group(container)
  File "/usr/local/lib/python3.8/site-packages/prefect_azure/container_instance.py", line 542, in _configure_container_group
    resource_group = resource_group_client.resource_groups.get(
  File "/usr/local/lib/python3.8/site-packages/azure/core/tracing/decorator.py", line 76, in wrapper_use_tracer
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/azure/mgmt/resource/resources/v2022_09_01/operations/_operations.py", line 11671, in get
    pipeline_response: PipelineResponse = self._client._pipeline.run(  # pylint: disable=protected-access
  File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/_base.py", line 202, in run
    return first_node.send(pipeline_request)
  File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/_base.py", line 70, in send
    response = self.next.send(request)
  File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/_base.py", line 70, in send
    response = self.next.send(request)
  File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/_base.py", line 70, in send
    response = self.next.send(request)
  [Previous line repeated 2 more times]
  File "/usr/local/lib/python3.8/site-packages/azure/mgmt/core/policies/_base.py", line 46, in send
    response = self.next.send(request)
  File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/policies/_redirect.py", line 156, in send
    response = self.next.send(request)
  File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/policies/_retry.py", line 448, in send
    response = self.next.send(request)
  File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/policies/_authentication.py", line 111, in send
    self.on_request(request)
  File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/policies/_authentication.py", line 91, in on_request
    self._update_headers(request.http_request.headers, self._token.token)
AttributeError: 'coroutine' object has no attribute 'token'```

**Comment by bjorhn** [comment]
Everything works after #106.

Thank you!
